src/wacks4square/square.py

```diff
@@ -0,0 +1,209 @@
+                print("DELETED", product["item_data"]["name"])
+                continue
+            variation_str = product["item_data"]["name"]
+            for variation in product.get("item_data", {}).get("variations", []):
+                variation_str += (
+                    f'|{variation["item_variation_data"]["name"]}<{variation["id"]}>'
+                )
+            all_variations.append(variation_str)
+        print("\n\n".join(all_variations))
+
+    def _get_orders_since_timestamp(self, timestamp):
+        end_at = datetime.utcnow().isoformat()
+        params = {
+            "limit": 500,
+            "location_ids": LOCATION_IDS,
+            # return full order data, not just order id
+            "return_entries": False,
+            "query": {
+                "filter": {
+                    "date_time_filter": {
+                        "closed_at": {"start_at": timestamp, "end_at": end_at}
+                    },
+                    "state_filter": {"states": ["COMPLETED"]},
+                },
+                # there seems to be some implicit hard limits on the square API, so to get the real latest timestamp
+                # we need to sort by desc to ensure the most recent items aren't truncated
+                "sort": {"sort_field": "CLOSED_AT", "sort_order": "DESC"},
+            },
+        }
+        new_orders_response = self._search_orders(params)
+        return new_orders_response
+
+    def get_sales_since_timestamp(self, timestamp: Optional[str]) -> List[Sale]:
+        """
+        Query the square API for all orders since the given timestamp. Then transform these orders
+        into a list of Sale objects.
+
+        last_timestamp datetime string in isoformat, may be null if wacks is running for the first time
+        """
+        logger.info(f"getting num sales, timestamp {timestamp}")
+        if not timestamp:
+            timestamp = self._get_two_years_ago_in_isoformat()
+        new_orders_response = self._get_orders_since_timestamp(timestamp)
+        sales = []
+        for order in new_orders_response:
+            line_items = order.get("line_items", [])
+            for item in line_items:
+                listing_id = item.get("catalog_object_id")
+                if not listing_id:
+                    continue
+                created_at = order["created_at"]
+                # some have milliseconds and some do not so let's strip the milliseconds out
+                created_at = re.sub(r"\..+Z", "", created_at)
+                # remove Z at the end for consistency
+                created_at = created_at.replace("Z", "")
+                try:
+                    sale_time = datetime.strptime(created_at, SQUARE_TIME_FORMAT)
+                except ValueError as e:
+                    logger.error(
+                        f"error converting order timestamp for order: {order}, item: {item}"
+                    )
+                    logger.error(e)
```

Comment:
I think if you do `logger.exception([msg])` it'll print the stack trace too, without having to `logger.error` twice